File: admin/src/services/appUser.ts
import { request } from './request';

const BASE_URL = '/api/v2/app-users';

export interface AppUser {
  id: number;
  phone: string;
  nickname?: string;
  avatar?: string;
  gender?: 'male' | 'female' | 'other';
  birth_date?: string;
  is_active: boolean;
  created_at: string;
  last_login?: string;
}

export interface AppUserListResponse {
  items: AppUser[];
  total: number;
  page: number;
  size: number;
}

export interface AppUserQueryParams {
  page?: number;
  size?: number;
  phone?: string;
  is_active?: boolean;
}

export const appUserService = {
  getAppUsers: async (params: AppUserQueryParams = {}) => {
    const response = await request.get<AppUserListResponse>(BASE_URL, { params });
    return response.data;
  },

  getAppUser: async (id: number) => {
    const response = await request.get<AppUser>(`${BASE_URL}/${id}`);
    return response.data;
  },

  updateAppUser: async (id: number, data: Partial<AppUser>) => {
    const response = await request.put<AppUser>(`${BASE_URL}/${id}`, data);
    return response.data;
  },

  deleteAppUser: async (id: number) => {
    await request.delete(`${BASE_URL}/${id}`);
  },
};

File: backend/app/models/app_user.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Boolean, DateTime, Date
from app.db.base_class import Base

class AppUser(Base):
    __tablename__ = "app_users"

    id = Column(Integer, primary_key=True, index=True)
    phone = Column(String, unique=True, index=True, nullable=False)
    nickname = Column(String, nullable=True)
    avatar = Column(String, nullable=True)
    gender = Column(String, nullable=True)  # male, female, other
    birth_date = Column(Date, nullable=True)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    last_login = Column(DateTime, nullable=True)
